Library/Homebrew/cmd/commands.rb

```diff
@@ -8,51 +8,69 @@ module Homebrew
   module_function
 
   def commands
-    if ARGV.include? "--quiet"
-      cmds = internal_commands + external_commands
-      cmds += internal_developer_commands
-      cmds += HOMEBREW_INTERNAL_COMMAND_ALIASES.keys if ARGV.include? "--include-aliases"
-      puts Formatter.columns(cmds.sort)
-    else
-      # Find commands in Homebrew/cmd
-      puts "Built-in commands"
-      puts Formatter.columns(internal_commands)
-
-      # Find commands in Homebrew/dev-cmd
-      puts
-      puts "Built-in developer commands"
-      puts Formatter.columns(internal_developer_commands)
-
-      # Find commands in the path
-      unless (exts = external_commands).empty?
+    command = CommandsCommand.new
+    command.call
+  end
+
+  class CommandsCommand < Command
+    def initialize
+      super
+      desc "Show a list of built-in and external commands"
```

Comment:
Instead of having this in`initialize` I think these should be class-wide method calls. Alternatively, `command "commands" do` or similar.